src/index.js

```diff
@@ -28,53 +31,74 @@ addInput.addEventListener('keypress', (event) => {
   }
 });
 
+clearAll.addEventListener('click', async () => {
+  const removeCompleted = arr.filter((arr) => arr.completed === false);
+  arr = removeCompleted;
+  defaultExport();
+});
+
 export function component() {
-  arr.forEach((card, n) => {
-    const element = document.createElement('li');
-    element.setAttribute('id', `i${card.index}`);
-    element.innerHTML += `
-      <input type="checkbox" name="task${card.index}">
+  if (Array.isArray(arr) && arr.length > 0) {
+    arr.forEach(async (card, n) => {
+      const element = document.createElement('li');
+      element.setAttribute('id', `i${card.index}`);
+      element.innerHTML += `
+      <input id="check${n}" type="checkbox" name="task${card.index}">
       <label for="task${card.index}">${card.description}</label>
       <textarea  id="task${n}" name="inputT" class="inputT">${card.description}</textarea>
       <button type="button" id="removeTask">
       <span id='delete-${n}' class="material-symbols-outlined  delete-ic">delete</span>
       <span class="material-symbols-outlined ic-vDots">more_vert</span>
       </button>
     `;
-    element.addEventListener('mouseenter', () => {
-      const deleteIC = document.getElementById(`delete-${n}`);
-      if (deleteIC) {
-        deleteIC.style.display = 'block';
-      }
-      element.querySelector('.ic-vDots').style.display = 'none';
-      element.style.backgroundColor = 'yellow';
-    });
-    element.addEventListener('mouseleave', () => {
-      const deleteIC = document.getElementById(`delete-${n}`);
-      if (deleteIC) {
-        deleteIC.style.display = 'none';
-      }
-      element.querySelector('.ic-vDots').style.display = 'block';
-      element.style.backgroundColor = 'white';
-    });
-    element.querySelector('#removeTask').addEventListener('click', async () => {
-      arrayIndex = `${card.index}` - 1;
-      const { deletingTask } = await import('./interactions.js');
-      deletingTask();
-      element.querySelector('.ic-vDots').style.display = 'none';
-    });
-    const z = element.querySelector('.inputT');
-    z.addEventListener('keypress', async (event) => {
-      if (event.key === 'Enter') {
-        event.preventDefault();
-        arr[n].description = z.value;
-        saveToLocalStorage();
-        removeAllTasks(container);
-        component();
-      }
+      element
+        .querySelector(`#check${n}`)
+        .addEventListener('click', async () => {
+          const sta = element.querySelector(`#check${n}`).checked;
+          if (sta) {
+            arr[n].completed = true;
+          } else {
+            arr[n].completed = false;
+          }
+        });
```

Comment:
- [ ] Great job working on the functionality so far but I think you didn't implement the requested changes by the previous code reviewer since when I press on checkox it doesn't change to ture if it is pressed or it is not it is keep false so whatever the user press on checkbox the completed turn into ture and the opposite so the idea like toggling `it will contain methods related to the status updates (completed: true / false).` so kindly work on it to make it change on the localstorage to true or false 🔥✨✅💯

## After pressing the checkbok it doesn't change to true ⬇⬇

![image](https://user-images.githubusercontent.com/58553711/229324926-df84a1e0-8391-4e20-adc5-36552f4a8adb.png)